### PR TITLE
Add magma

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge/staged-recipes, this is ready. I'll remove the azure changes after a review.

File: recipes/magma/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  skip: True  # [not linux64]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    # TODO: fix this in the feedstock
+    # - {{ compiler('cuda') }}
+    - nvcc_linux-64 9.2
+    - cmake
```
Comment (chrisburr) [inline recipes/magma/meta.yaml:26]:
```suggestion
    - cmake
    - make
```

File: recipes/magma/thread_queue.patch [hunk 2]
```diff
@@ -0,0 +1,20 @@
+--- control/thread_queue.cpp	2016-08-30 06:37:49.000000000 -0700
++++ control/thread_queue.cpp	2016-10-10 19:47:28.911580965 -0700
+@@ -15,7 +15,7 @@
+ {
+     if ( err != 0 ) {
+         fprintf( stderr, "Error: %s (%d)\n", strerror(err), err );
+-        throw std::exception();
++        // throw std::exception();
```
Comment (chrisburr) [inline recipes/magma/thread_queue.patch:8]: What’s the reason for this patch? It looks like it just changes the functionality of the package

Comment (isuruf) [reply recipes/magma/thread_queue.patch:8]: I'm not sure. I took this recipe from AnacondaRecipes and pytorch recipes and both have it. I'd rather keep it as they know what they are doing and this package is needed for pytorch.

Comment (chrisburr) [reply recipes/magma/thread_queue.patch:8]: That’s good enough for me

**Review by chrisburr — APPROVED** [review]
Looks good to me once the CI changes are reverted

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ jobs:
     matrix:
       linux:
         CONFIG: linux64
-        IMAGE_NAME: quay.io/condaforge/linux-anvil-comp7
+        IMAGE_NAME: quay.io/condaforge/linux-anvil-cuda:9.2
```
Comment (ocefpaf) [inline .azure-pipelines/azure-pipelines-linux.yml:11]: We need a smarter way to do this to avoid setting and resetting it.

**Review by ocefpaf — APPROVED** [review]

**Comment by isuruf** [comment]
Thanks for the reviews

**Comment by jakirkham** [comment]
Thanks Isuru! 😄

**Pull request merged**